### PR TITLE
add checks when unpacking spring

### PR DESCRIPTION
**Pull request by moonso**
If using `--first-checksum` and `--second-checksum` in `crunchy decompress spring` it will check the integrity of the decompressed fastq files.

### CODE REVIEW

File: crunchy/cli/compare_cmd.py [hunk 1]
```diff
@@ -11,19 +11,34 @@
...
+@click.option(
+    "--checksum", "-c", help="If the file should be compared to a checksum directly",
+)
 @click.option(
     "--algorithm", "-a", type=click.Choice(["md5", "sha1", "sha256"]), default="sha256"
 )
 @click.option("--dry-run", is_flag=True)
-def compare(first, second, algorithm, dry_run):
-    """Compare two files by generating checksums. Fails if two files differ."""
+def compare(first, second, algorithm, checksum, dry_run):
```
Comment (henrikstranneheim) [inline crunchy/cli/compare_cmd.py:22]:
```suggestion
def compare_checksums(first, second, algorithm, checksum, dry_run):
```

File: crunchy/cli/compare_cmd.py [hunk 2]
```diff
@@ -11,19 +11,34 @@
 
 @click.command()
```
Comment (henrikstranneheim) [inline crunchy/cli/compare_cmd.py:12]:
```suggestion
@click.command("compare-checksums")
```
Would be nice to know what we are comparing

File: crunchy/cli/decompress_cmd.py [hunk 3]
```diff
@@ -19,23 +20,35 @@ def decompress():
 @click.command()
 @click.argument("spring-path", type=click.Path(exists=True))
 @click.option(
-    "--first", "-f", type=click.Path(exists=False), help="First read in pair",
+    "--first",
+    "-f",
+    type=click.Path(exists=False),
+    help="Fastq file that spring will be decompressed to. First read in pair",
```
Comment (henrikstranneheim) [inline crunchy/cli/decompress_cmd.py:26]:
```suggestion
    help="Fastq file that spring archive will be decompressed to. First read in pair",
```

File: crunchy/cli/decompress_cmd.py [hunk 4]
```diff
@@ -19,23 +20,35 @@ def decompress():
...
 )
 @click.option(
-    "--second", "-s", type=click.Path(exists=False), help="Second read in pair",
+    "--second",
+    "-s",
+    type=click.Path(exists=False),
+    help="Fastq file that spring will be decompressed to. Second read in pair",
```
Comment (henrikstranneheim) [inline crunchy/cli/decompress_cmd.py:32]:
```suggestion
    help="Fastq file that spring archive will be decompressed to. Second read in pair",
```

File: crunchy/cli/compare_cmd.py [hunk 5]
```diff
@@ -11,19 +11,34 @@
...
+        raise click.Abort
+
     if dry_run:
         LOG.info("Dry Run!")
+
     checksums = []
+    if checksum:
+        checksums.append(checksum)
 
     for _infile in [first, second]:
```
Comment (henrikstranneheim) [inline crunchy/cli/compare_cmd.py:39]: Why do you need second here?

File: crunchy/cli/decompress_cmd.py [hunk 6]
```diff
@@ -19,23 +20,35 @@ def decompress():
 @click.command()
 @click.argument("spring-path", type=click.Path(exists=True))
 @click.option(
-    "--first", "-f", type=click.Path(exists=False), help="First read in pair",
+    "--first",
```
Comment (henrikstranneheim) [inline crunchy/cli/decompress_cmd.py:23]:
```suggestion
    "--first",
```
First seems so unspecific maybe `first_read`or even `first_read_file`. I know it is longer, but you have to think a lot less 😄 . I saw in the test that you use first_read sometimes

File: crunchy/cli/compare_cmd.py (hunk 1 above)
Comment (moonso) [reply crunchy/cli/compare_cmd.py:22]: No I want the cli commands to be short! But I can expand the description

File: crunchy/cli/compare_cmd.py (hunk 5 above)
Comment (moonso) [reply crunchy/cli/compare_cmd.py:39]: Sometimes we compare two fastq files and sometimes we compare one fastq file with a checksum

File: crunchy/cli/decompress_cmd.py (hunk 6 above)
Comment (moonso) [reply crunchy/cli/decompress_cmd.py:23]: Sure

File: crunchy/cli/compare_cmd.py (hunk 5 above)
Comment (henrikstranneheim) [reply crunchy/cli/compare_cmd.py:39]: 👍

File: crunchy/cli/compare_cmd.py (hunk 1 above)
Comment (henrikstranneheim) [reply crunchy/cli/compare_cmd.py:22]: Fair enough 😄

**Pull request merged**